File: AspNetMvcUnleashed/0672329980_source/Chapter07/VB/MvcApplication1/MvcApplication1/Controllers/MovieController.vb
Public Class MovieController
    Inherits System.Web.Mvc.Controller

    Private _entities As New MoviesDBEntities()

    '
    ' GET: /Movie/

    Function Index() As ActionResult
        Return View(_entities.MovieSet.ToList())
    End Function


    '
    ' GET: /Movie/Create

    Function Create() As ActionResult
        Return View()
    End Function


    ' POST: /Movie/Create
    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(<Bind(Exclude:="Id")> ByVal movieToCreate As Movie) As ActionResult
        If Not ModelState.IsValid Then
            Return View()
        End If

        ' Add movie to database
        Return RedirectToAction("Index")
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter07/CS/MvcApplication1/MvcApplication1/Controllers/MovieController.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class MovieController : Controller
    {
        private MoviesDBEntities _entities = new MoviesDBEntities();

        //
        // GET: /Movie/

        public ActionResult Index()
        {
            return View(_entities.MovieSet.ToList());
        }

        //
        // GET: /Movie/Create

        public ActionResult Create()
        {
            return View();
        } 

        //
        // POST: /Movie/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create([Bind(Exclude="Id")]Movie movieToCreate)
        {
            if (!ModelState.IsValid)
                return View();

            // Add movie to database
            return RedirectToAction("Index");
        }


    }
}
